Gateway/Http/Client/TransactionPosCloudSync.php

```diff
@@ -0,0 +1,88 @@
+ * ############    ####( ######  #####. ######  ############   ############
+ * #############  #####( ######  #####. ######  #############  #############
+ *        ######  #####( ######  #####. ######  #####  ######  #####  ######
+ * ###### ######  #####( ######  #####. ######  #####  #####   #####  ######
+ * ###### ######  #####( ######  #####. ######  #####          #####  ######
+ * #############  #############  #############  #############  #####  ######
+ *  ############   ############  #############   ############  #####  ######
+ *                                      ######
+ *                               #############
+ *                               ############
+ *
+ * Adyen Payment Module
+ *
+ * Copyright (c) 2018 Adyen B.V.
+ * This file is open source and available under the MIT license.
+ * See the LICENSE file for more info.
+ *
+ * Author: Adyen <magento@adyen.com>
+ */
+
+
+namespace Adyen\Payment\Gateway\Http\Client;
+
+
+use Magento\Payment\Gateway\Http\ClientInterface;
+
+class TransactionPosCloudSync implements ClientInterface
+{
+
+    public function __construct(
+        \Magento\Framework\Model\Context $context,
+        \Magento\Framework\Encryption\EncryptorInterface $encryptor,
+        \Adyen\Payment\Helper\Data $adyenHelper,
+        \Adyen\Payment\Logger\AdyenLogger $adyenLogger,
+        \Adyen\Payment\Model\RecurringType $recurringType,
+        array $data = []
+    ) {
+        $this->_encryptor = $encryptor;
+        $this->_adyenHelper = $adyenHelper;
+        $this->_adyenLogger = $adyenLogger;
+        $this->_recurringType = $recurringType;
+        $this->_appState = $context->getAppState();
+
+        // initialize client
+        $apiKey = $this->_adyenHelper->getApiKey();
+        $client = new \Adyen\Client();
+        $client->setApplicationName("Magento 2 plugin");
+        $client->setXApiKey($apiKey);
+        if ($this->_adyenHelper->isDemoMode()) {
+            $client->setEnvironment(\Adyen\Environment::TEST);
+        } else {
+            $client->setEnvironment(\Adyen\Environment::LIVE);
+        }
+        // assign magento log
+        $client->setLogger($adyenLogger);
+
+        $this->_client = $client;
+    }
+
+    /**
+     * Places request to gateway. Returns result as ENV array
+     *
+     * @param \Magento\Payment\Gateway\Http\TransferInterface $transferObject
+     * @return array
+     * @throws \Magento\Payment\Gateway\Http\ClientException
+     * @throws \Magento\Payment\Gateway\Http\ConverterException
+     */
+    public function placeRequest(\Magento\Payment\Gateway\Http\TransferInterface $transferObject)
+    {
+        $request = $transferObject->getBody();
+        $service = new \Adyen\Service\PosPayment($this->_client);
+        $transactionType = \Adyen\TransactionType::GOODS_SERVICES;
+        $poiId = $this->_adyenHelper->getPoiId();
+        $json = $service->getPaymentRequest($poiId, $request['amount']['value'], $request['amount']['currency'],
```

Comment:
service should not be used for creating a paymentRequest. You provide a request to the service and this does the request. Move this or to a Util class or even better create a runTenderSync where you provide the variables and this creates the request and execute it. This reduces the complexity to one call.